com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/util/HooksHelper.java

```diff
@@ -0,0 +1,161 @@
+import com.sap.cloud.lm.sl.cf.core.cf.metadata.processor.EnvMtaMetadataParser;
+import com.sap.cloud.lm.sl.cf.core.cf.metadata.processor.MtaMetadataParser;
+import com.sap.cloud.lm.sl.cf.core.cf.metadata.util.MtaMetadataUtil;
+import com.sap.cloud.lm.sl.cf.core.model.DeployedMtaApplication;
+import com.sap.cloud.lm.sl.cf.core.model.HookPhase;
+import com.sap.cloud.lm.sl.cf.process.steps.ProcessContext;
+import com.sap.cloud.lm.sl.cf.process.steps.StepPhase;
+import com.sap.cloud.lm.sl.cf.process.steps.StepsUtil;
+import com.sap.cloud.lm.sl.cf.process.variables.Variables;
+import com.sap.cloud.lm.sl.mta.model.DeploymentDescriptor;
+import com.sap.cloud.lm.sl.mta.model.Hook;
+import com.sap.cloud.lm.sl.mta.model.Module;
+
+public class HooksHelper {
+
+    private final MtaMetadataParser mtaMetadataParser;
+    private final EnvMtaMetadataParser envMtaMetadataParser;
+    private final ProcessContext context;
+    private final HookPhase hookPhaseBeforeStep;
+    private final HookPhase hookPhaseAfterStep;
+
+    private HooksHelper(Builder hooksHelperBuilder) {
+        this.mtaMetadataParser = hooksHelperBuilder.mtaMetadataParser;
+        this.envMtaMetadataParser = hooksHelperBuilder.envMtaMetadataParser;
+        this.context = hooksHelperBuilder.context;
+        this.hookPhaseBeforeStep = hooksHelperBuilder.hookPhaseBeforeStep;
+        this.hookPhaseAfterStep = hooksHelperBuilder.hookPhaseAfterStep;
+    }
+
+    public List<Hook> getHooksForStepPhase(Module moduleToDeploy, StepPhase currentStepPhase) {
+        HookPhase currentHookPhaseForExecution = determineHookPhaseForCurrentStepPhase(currentStepPhase);
+        return getHooksForCurrentPhase(moduleToDeploy, currentHookPhaseForExecution);
+    }
+
+    public Module determineModuleToDeploy(ProcessContext context) {
+        Module moduleToDeploy = context.getVariable(Variables.MODULE_TO_DEPLOY);
+        return moduleToDeploy != null ? moduleToDeploy : determineModule(context);
+    }
+
+    public void setHooksForExecution(List<Hook> hooksForExecution) {
+        context.setVariable(Variables.HOOKS_FOR_EXECUTION, hooksForExecution);
+    }
+
+    private Module determineModule(ProcessContext context) {
+        DeploymentDescriptor deploymentDescriptor = context.getVariable(Variables.COMPLETE_DEPLOYMENT_DESCRIPTOR);
+        if (deploymentDescriptor == null) {
+            // This will be the case only when the process is undeploy.
+            return null;
+        }
+        return determineModuleFromDeploymentDescriptor(context.getVariable(Variables.APP_TO_PROCESS), deploymentDescriptor);
+    }
+
+    private Module determineModuleFromDeploymentDescriptor(CloudApplicationExtended cloudApplication,
+                                                           DeploymentDescriptor deploymentDescriptor) {
+        HandlerFactory handlerFactory = StepsUtil.getHandlerFactory(context.getExecution());
+        if (cloudApplication.getModuleName() == null) {
```

Comment:
I was thinking more along the lines of
<pre>
Module determineModuleFromDescriptor(...){
   HandlerFactory handlerFactory = getHandlerFactory();
   String moduleName = computeModuleName(...);
   return findModuleByName(...);
}

String computeModuleName(...) {
   if (cloudApp.getModuleName() == null {
      return parseModuleName(); //use getDeployedMtaApplication().getModuleName() here
   }
   return cloudApp.getModuleName();
}
</pre>

But your solution is good as well

Reply:
Looks better. Done